sdks/python/apache_beam/examples/snippets/snippets.py

```diff
@@ -502,6 +502,52 @@ def process(self, element):
     p.visit(SnippetUtils.RenameFiles(renames))
 
 
+def examples_wordcount_templated(renames):
+  """Templated WordCount example snippet."""
+  import re
+
+  import apache_beam as beam
+  from apache_beam.io import ReadFromText
+  from apache_beam.io import WriteToText
+  from apache_beam.options.pipeline_options import PipelineOptions
+
+  # [START example_wordcount_templated]
+  class WordcountOptions(PipelineOptions):
+    @classmethod
+    def _add_argparse_args(cls, parser):
+      # Use add_value_provider_argument for arguments to be templatable
+      # Use add_argument as usual for non-templatable arguments
```

Comment:
I see what you mean now (just move the 2nd line down). I meant for it to be this way, so that I can explain both before using them, to make more obvious the contrast...